neuralmonkey/trainers/generic_trainer.py

```diff
@@ -39,8 +45,23 @@ def __init__(self, objectives: List[Objective],
         # Flatten the list of lists
         self.var_list = [var for var_list in var_lists for var in var_list]
 
+        if decay_function is None:
+            decay_function = constant_decay()
+
         with tf.name_scope("trainer"):
-            self.optimizer = optimizer or tf.train.AdamOptimizer(1e-4)
+            if global_step is None:
+                global_step = tf.Variable(
+                    0, trainable=False, name="global_step")
+            self.global_step = global_step
+
+            self.optimizer = optimizer_getter(  # type: ignore
+                self.global_step,
+                decay_function)
+            # TODO: avoid accessing private member
+            # pylint: disable=protected-access
+            tf.summary.scalar("learning_rate", self.optimizer._lr,
```

Comment:
Kreslil bych ten, jenom když ten lr bude tensor, když bude float, tak bych na to kašlal. Už tahle jsou ty events fily obrovský, nepotřebujeme tam psát konstantu.